### DISCUSSION TITLE
Cache messages/headers when doing direct search

### DISCUSSION THREAD

**Discussion by Miwer** [Tensai75/nzb-monkey-go#23] [Q&A] [answered]
Hi - Whenever I do a direct search against my usual usenet server, I find it takes a couple of minutes each time, because of searching messages/headers +/-24 hours. 
When I have multiple posts to search for within the same day, I believe it still downloads the same headers for that time period from the server each time. Correct me if I'm wrong.
Is it possible to make some kind of cache function (perhaps configurable to max size or whatever), to keep the headers locally, as to increase the search time, when they have already been downloaded once?
If that's not how the direct search function works, then please excuse my ignorance :D

**Comment by Tensai75** [comment, marked as answer]
In theory, your idea is not wrong. However, I doubt the speed increase would be very significant in practice.
You can only benefit from a cache if the subsequent searches are done in the same group(s) and for the same time periods for each group. This is already a very specific requirement. And even if this was the case, on an average modern computer with an average internet download speed, scanning the loaded headers takes much longer than the download step. And so even in this case the speed increase with a cache would probably be less than 10-20%.

Therfore, and because such a cache function would require a lot of changes to the code, I don't think it is worth the effort. Maybe someone else would like to do it. But not me :-)

Regards, Tensai
